kpatch-build/kpatch-build

```diff
@@ -884,6 +907,10 @@ remove_patches
 
 cp -LR "$DATADIR/patch" "$TEMPDIR" || die
 
+if [[ "$KLP_REPLACE" -eq 1 ]] ; then
+   sed -i 's/lpatch->replace = false/lpatch->replace = true/g' "$TEMPDIR/patch/livepatch-patch-hook.c"
+fi
+
```

Comment:
I don't know how I feel about this approach -- in a way, atomic replace and other features that directly affect the `klp_patch` object are more than just build options, they may change the way the kpatch is coded.  I wonder if we should try to move this in the kpatch itself somehow?  This might be too difficult to implement (how to check that said feature is supported, etc.) but I ask in case others have any good ideas.

Reply:
With current klp_patch->replace design, which unloads **all** loaded patch with the new patch, I think the decision to use replace or not is done at kpatch-build time. Therefore, we don't need much flexibility in kpatch. (I may have missed some meaningful scenarios, though. )

Reply:
Right, the question I have is whether `--replace` might tempt somebody to use the same input patch without the flag.  If `klp_patch->replace` behavior is encoded in the kpatch itself, then the kpatch author be responsible for doing so.   That is to say, the patch author wrote it in the style for atomic replace and expects it to be built that way.

Reply:
Song - for your kpatch infrastructure, would you be intending to build _all_ kpatches with atomic replace?  I ask, as maybe another way to look at this is to default all kpatches as `klp->replace` where available.  Just curious what your use case is.

Reply:
In our use cases, I would like to build all kpatches with atomic replace. Each host will have at most one kpatch load at anytime. I think this would give fewer confusions.